syscall/iov.c

```diff
@@ -151,19 +151,22 @@ int oe_iov_sync(
             if (n < buf[i].iov_len)
                 goto done;
 
+            /* Fail if offset value stored in iov_base is too large */
+            if (buf_size < (size_t)buf[i].iov_base)
+                goto done;
+
             /* Sync the base data for this element. */
             {
                 /* Note: buf[i].iov_base is an offset here (not a pointer). */
-                uint8_t* src = (uint8_t*)buf[i].iov_base + (uint64_t)buf;
+                uint8_t* src = (uint8_t*)buf + (uint64_t)buf[i].iov_base;
```

Comment:
why does this make a difference?

Reply:
I think its just to have a better style, like point first then offset. 

`new_ptr = ptr + offset`

Reply:
This doesn't make any difference, it just makes the code conceptually clearer because buf[i].iov_base is stored as an offset in oe_iov_pack, so it makes sense to cast it as a uint64_t here instead to represent the offset